x-pack/plugins/maps/public/classes/sources/esql_source/esql_source.tsx

```diff
@@ -202,6 +206,11 @@ export class ESQLSource
       const timeFilter = getTime(undefined, timeRange, {
         fieldName: this._descriptor.dateField,
       });
+      const namedParams = getEarliestLatestParams(this._descriptor.esql, timeRange);
```

Comment:
It looks like the problem I flagged is happening because `getEarliestLatestParams` is in `if (requestMeta.applyGlobalTime)` block. Instead, `getEarliestLatestParams` should be moved outside of this if statement so params are always added.

I would assume users would disable "Apply global time range" when using `?earliest` and `?latest` since you would not want both.

Reply:
Great find Nathan! I had to move the timeRange computation outside too. I hope it makes sense https://github.com/elastic/kibana/pull/184361/commits/04a243f2b45ccbe70c2d8e232009ff8ce450f569